cli/exp_rpty.go

```diff
@@ -0,0 +1,217 @@
+package cli
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/mattn/go-isatty"
+	"golang.org/x/term"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/cliui"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/workspacesdk"
+	"github.com/coder/coder/v2/pty"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) rptyCommand() *serpent.Command {
+	var (
+		client = new(codersdk.Client)
+		args   handleRPTYArgs
+	)
+
+	cmd := &serpent.Command{
+		Handler: func(inv *serpent.Invocation) error {
+			if r.disableDirect {
+				return xerrors.New("direct connections are disabled, but you can try websocat ;-)")
```

Comment:
:)

Reply:
I had to actually try it out and it _does_ work! Except you have to enter raw JSON objects to write to the terminal...

Reply:
This is very convenient for debugging purposes. Leave the websocat hint please!